app/Controllers/LibriController.php

```
@@ -1195,14 +1195,19 @@ public function store(Request $request, Response $response, mysqli $db): Respons
+            // a request for three copies must never leave a partial 1/3 or 2/3
+            // result if one inventory code fails. Codes stay uniform (-C1, -C2,
+            // ...) and collision-free through the repository allocator.
+            $createdCopies = $copyRepo->createManyForBook(
+                $id,
+                $baseInventario,
+                $copieTotali,
+                'disponibile',
+                __('Copia %d di %d')
+            );
+            if ($createdCopies !== $copieTotali) {
+                throw new \RuntimeException('Unable to create the requested physical copies.');
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '1210', column '27'). (undefined)

(MissingImport)